scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,434 @@
+func (o EvidenceHandler) GetName() string {
+	return "sevsnp-evidence-handler"
+}
+
+// GetAttestationScheme returns the attestation scheme
+func (o EvidenceHandler) GetAttestationScheme() string {
+	return SchemeName
+}
+
+// GetSupportedMediaTypes returns the supported media types for the SEVSNP scheme
+func (o EvidenceHandler) GetSupportedMediaTypes() []string {
+	return EvidenceMediaTypes
+}
+
+// ExtractClaims converts evidence in tsm-report format to our
+// "internal representation", which is in CoRIM format.
+func (o EvidenceHandler) ExtractClaims(
+	token *proto.AttestationToken,
+	_ []string,
+) (map[string]interface{}, error) {
+	var claimsSet map[string]interface{}
+	var tsm tokens.TSMReport
+
+	err := tsm.FromCBOR(token.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	reportProto, err := abi.ReportToProto(tsm.OutBlob)
+	if err != nil {
+		return nil, err
+	}
+
+	refValComid, err := sevsnpParser.ReportToComid(reportProto, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	err = refValComid.Valid()
+	if err != nil {
+		return nil, err
+	}
+
+	refValCorim := corim.UnsignedCorim{}
+	refValCorim.SetProfile("http://amd.com/2024/snp-corim-profile")
+	refValCorim.AddComid(refValComid)
+
+	refValJson, err := refValCorim.ToJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(refValJson, &claimsSet)
+	if err != nil {
+		return nil, err
+	}
+
+	return claimsSet, nil
+}
+
+// snpAttestationOptions parameter for verifying certificate chain
+func snpAttestationOptions() *verify.Options {
+	return &verify.Options{
+		Getter:              trust.DefaultHTTPSGetter(),
+		Now:                 time.Now(),
+		DisableCertFetching: true,
+	}
+}
+
+// readCert helper function to read a certificate from a blob
+func readCert(cert []byte) ([]byte, error) {
+	block, _ := pem.Decode(cert)
+	if block == nil || block.Type != "CERTIFICATE" {
+		return nil, fmt.Errorf("failed to read certificate")
+	}
+	return block.Bytes, nil
+}
+
+// compareTAs compares two X509 certificates for equality
+func compareTAs(provisionedArk []byte, evidenceArk []byte) (bool, error) {
+	pArk, err := readCert(provisionedArk)
+	if err != nil {
+		return false, err
+	}
+
+	pCert, err := x509.ParseCertificate(pArk)
+	if err != nil {
+		return false, err
+	}
+
+	eArk, err := readCert(evidenceArk)
+	if err != nil {
+		return false, err
+	}
+
+	eCert, err := x509.ParseCertificate(eArk)
+	if err != nil {
+		return false, err
+	}
+
+	return pCert.Equal(eCert), nil
+}
+
+// ValidateEvidenceIntegrity verifies that the ARK in the
+// evidence matches the provisioned ARK, confirms the
+// integrity of the certificate chain, and validates
+// the signature of the evidence.
+//
+// The "auxblob" in the evidence contains a certificate chain.
+// The Trust Anchor in this chain is AMD Root Key (ARK).
+func (o EvidenceHandler) ValidateEvidenceIntegrity(
+	token *proto.AttestationToken,
+	trustAnchors []string,
+	endorsementsStrings []string,
```

Comment:
Why are endorsements strings instead of CoMIDs for example? This is also not used, so should be `_`.

Reply:
The [IEvidenceHandler](https://github.com/veraison/services/blob/main/handler/ievidencehandler.go) interface defines it that way.